core/src/main/scala/org/apache/spark/MapOutputTracker.scala

```diff
@@ -1011,4 +1039,15 @@ private[spark] object MapOutputTracker extends Logging {
 
     splitsByAddress.mapValues(_.toSeq).iterator
   }
+
+  def checkMapStatuses(statuses: Array[MapStatus], shuffleId: Int): Unit = {
+    assert (statuses != null)
+    for (status <- statuses) {
+      if (status == null) {
```

Comment:
I think you can extract this `if` into a new method and reuse the method in `convertMapStatuses`.